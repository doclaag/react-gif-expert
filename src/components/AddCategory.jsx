import { useState } from 'react';
import  PropTypes  from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    const newInputValue = inputValue.trim();

    if (newInputValue.length <= 1) return;

    setInputValue('');
    onNewCategory(newInputValue);
  };

  return (
    <form onSubmit={onSubmit} className="mb-4 bg-slate-800 rounded-md p-4">
      <input
        type="text"
        placeholder="Buscar Gifs..."
        value={inputValue}
        onChange={onInputChange}
        className="border-2 border-gray-300 rounded-md px-2 py-1 focus:outline-none focus:border-blue-500"
      />
    </form>
  );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired, // onNewCategory debe ser una funci√≥n y es requerida
  };
