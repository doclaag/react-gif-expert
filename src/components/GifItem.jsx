import PropTypes from "prop-types";



export const GifItem = ({ title, url }) => {

    // console.log(title, url)

    return (
        <div className="bg-white rounded-lg shadow-md">
      <div className="aspect-w-16 aspect-h-9">
        <img src={url} alt={title} className="object-cover w-full h-auto rounded-lg" />
      </div>
      <p className="text-center mt-4 font-semibold text-gray-800">{title}</p>
    </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
