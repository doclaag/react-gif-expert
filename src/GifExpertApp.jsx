import { useState } from "react";

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Eren Jaeger"]);

  const onAddCategory = (newCategory) => {
    if (
      !categories.findIndex(
        (category) => category.toLowerCase() === newCategory.toLowerCase()
      )
    )
      return;

    // console.log(newCategory)
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1 className="text-white text-2xl font-bold mb-4">GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
