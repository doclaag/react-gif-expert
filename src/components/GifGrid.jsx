import PropTypes from "prop-types";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);


    return (
        <>
            <div className="p-4 rounded-lg shadow-md">
                <h3 className="text-2xl font-bold text-white mb-4 text-center">{category}</h3>

                {isLoading && <p className="text-lg font-bold text-gray-800">Loading...</p>}

                <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-3 xl:grid-cols-3 gap-4">
                    {images.map((image) => (
                        <GifItem key={image.id} {...image} />
                    ))}
                </div>

            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
