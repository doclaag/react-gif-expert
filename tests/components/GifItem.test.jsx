const { render, screen } = require("@testing-library/react");
import { GifItem } from "../../src/components";

describe("Tests on <GifItem/>", () => {
  const title = "Saitama";
  const url = "https://one.punch.com/saitama.png";

  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("should show the image with the URL and ALT indicated", () => {
    render(<GifItem title={title} url={url} />);
    //  screen.debug();

    const {src, alt} = screen.getByRole("img");
    expect(src).toEqual(url);
    expect(alt).toEqual(title);
  });

  test('should show the title in the component', () => { 
    render(<GifItem title={title} url={url} />);

    const titleComponent = screen.getByText(title);
    expect(titleComponent).toBeTruthy();
   })
});
