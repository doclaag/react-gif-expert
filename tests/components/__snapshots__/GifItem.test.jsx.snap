// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem/> should match the snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-md"
  >
    <div
      class="aspect-w-16 aspect-h-9"
    >
      <img
        alt="Saitama"
        class="object-cover w-full h-auto rounded-lg"
        src="https://one.punch.com/saitama.png"
      />
    </div>
    <p
      class="text-center mt-4 font-semibold text-gray-800"
    >
      Saitama
    </p>
  </div>
</div>
`;
